src/puppetlabs/puppetdb/scf/storage_utils.clj

```diff
@@ -147,6 +147,24 @@
        (= (:relname (first results))
           index))))
 
+(pls/defn-validated constraint-exists? :- s/Bool
+  ([table :- s/Str
+    constraint :- s/Str]
+   (let [schema (current-schema)]
+     (constraint-exists? schema table constraint)))
+
+  ([schema :- s/Str
+    table :- s/Str
+    constraint :- s/Str]
+
+   (let [query (str "select count(*)"
+                    "  from information_schema.constraint_column_usage"
+                    "  where table_schema = ?"
+                    "  and table_name = ?"
+                    "  and constraint_name = ?")
+         results (jdbc/query-to-vec [query schema table constraint])]
```

Comment:
I just took it from like 146 above without thinking... and am happy to keep it if it works :)